---
title: "CC3"
output: github_document
author: LE COZ SARAH
---
```{r library, include=FALSE}
library(stringr)
library(rmarkdown)
library(knitr)
library(phyloseq)
library(dada2)
library(DECIPHER)
library(phangorn)
library(ggplot2)
library(gridExtra)
library(shiny)
library(miniUI)
library(caret)
library(pls)
library(e1071)
library(ggplot2)
library(randomForest)
library(dplyr)
library(ggrepel)
#library(nlme)
library(devtools)
library(reshape2)
library(PMA)
#library(structSSI)
library(ade4)
library(ggnetwork)
library(intergraph)
library(scales)
library(genefilter)
library(impute)
library(phyloseqGraphTest)
library(Biostrings)
library(RSQLite)
library(parallel)
library(ape)
```

# méthode de séquençage utilisée : Illumina HiSeq 2500 sequencing; 16S rRNA
# séquençage du microbiote du vagin et de l'utérus de femmes en bonne santé et de femmes atteintes d'endométriose.

```{bash, eval=FALSE}
mkdir data
wget -P data -i URL
```

```{bash}
cd data
gunzip -d *.gz
```

```{r path}
path <- "data"
list.files(path)
```

```{r}
# Creer une variable qui contient les donnees finissant par _R1 ou _R2 pour les trier par ordre alphabetique
fnFs <- sort(list.files(path, pattern="_R1.fastq", full.names=TRUE))
fnRs <- sort(list.files(path, pattern="_R2.fastq", full.names=TRUE))
# pour avoir le nom du fichier
sample.names <- sapply(strsplit(basename(fnFs), "_R"), `[`, 1)
```

```{r}
plotQualityProfile(fnFs[1:2])
```


```{r}
plotQualityProfile(fnRs[1:2])
```

```{r}
# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names
```